tests/apollo/test_skvbc_persistence.py

```diff
@@ -115,7 +116,17 @@ async def test_read_written_data_after_restart_of_all_nodes(self, bft_network, t
         """
         bft_network.start_all_replicas()
         skvbc = kvbc.SimpleKVBCProtocol(bft_network, tracker)
-        (key, val) = await skvbc.send_write_kv_set()
+
+        # Ensure execution happens on all replicas before shutdown
+        # by requiring an N of N quorum of replies.
+        # Since we're creating a single block on the chain we try to 
+        # avoid the state where some replica failed to execute the block
+        # and needs to recover by starting from the genesis block. A recovery with less than 2
+        # total blocks on the chain is unhandled by the system.
+        # For more info see BC-17607.
+
+        quorum = MofNQuorum.All(bft_network.config, bft_network.all_replicas())
+        (key, val) = await skvbc.send_write_kv_set(m_of_n_quorum=quorum)
```

Comment:
If you addressed my previous comment, this would read more naturally:
`send_write_kv_set(quorum=quorum)`